vocode/streaming/agent/state_agent.py

```diff
@@ -125,9 +126,10 @@ async def generate_completion(
 
         if self.resume:
             if human_input:
-                self.resume = await self.resume(human_input)
```

Comment:
I don't think this change is needed. IIRC it was so `self.resume` could update itself

I think this whole process will be easier to reason about if we have as many pure functions as possible, and avoid touching global state deep in the callstack